src/game_actor.cpp

```diff
@@ -325,16 +334,55 @@ void Game_Actor::AddState(int state_id) {
 
 void Game_Actor::RemoveState(int state_id) {
 	Game_Battler::RemoveState(state_id);
+	AddEquipmentStates();
 }
 
 void Game_Actor::RemoveBattleStates() {
 	Game_Battler::RemoveBattleStates();
+	AddEquipmentStates();
```

Comment:
Todo: AddEquipmentStates only needed in RemoveState, all code paths go through it (except RemoveAllStates, refactor?)

Reply:
Instead of removing and adding the states back, would it make more sense to just filter in the remove? Like `if (!isEquipState(state)) Game_Battler::RemoveState(state);`